certified-connectors/DocuSignDemo/script.csx

```diff
@@ -3007,6 +3083,340 @@ public class Script : ScriptBase
+      {
+        itemProperties[roleName + ":::Secondary Country Code"] = new JObject
+        {
+          ["type"] = "string",
+          ["x-ms-summary"] = roleName + " SMS Country Code"
+        };
+        itemProperties[roleName + ":::Secondary Phone Number"] = new JObject
+        {
+          ["type"] = "string",
+          ["x-ms-summary"] = roleName + " SMS Phone Number"
+        };
+      }
+      if (string.Equals(recipientObj["deliveryMethod"].ToString(), "SMS", StringComparison.OrdinalIgnoreCase))
+      {
+        itemProperties[roleName + ":::Country Code"] = new JObject
+        {
+          ["type"] = "string",
+          ["x-ms-summary"] = roleName + " SMS Country Code"
+        };
+        itemProperties[roleName + ":::Phone Number"] = new JObject
+        {
+          ["type"] = "string",
+          ["x-ms-summary"] = roleName + " SMS Phone Number"
+        };
+      }
+      else
+      {
+        if (string.Equals(recipientObj["recipientType"].ToString(), "signer"))
+        {
+          itemProperties[roleName + ":::Email"] = new JObject
+          {
+            ["type"] = "string",
+            ["x-ms-summary"] = roleName + " Recipient Email (Leave empty if there’s a signing group)"
+          };
+        }
+        else
+        {
+          itemProperties[roleName + ":::Email"] = new JObject
+          {
+            ["type"] = "string",
+            ["x-ms-summary"] = roleName + " Recipient Email"
+          };
+        }
+      }
+
+      // Tabs fields
+      JObject singleRecipientData = recipientObj as JObject;
+      JObject tabsData = singleRecipientData["tabs"] as JObject;
+      if (tabsData != null)
+      {
+        foreach (var tabType in editableTabs)
+        {
+          if (tabsData.ContainsKey(tabType))
+          {
+            foreach (var tab in tabsData[tabType] as JArray)
+            {
+              if (string.Equals(tabType, "checkboxTabs", StringComparison.OrdinalIgnoreCase))
+              {
+                itemProperties[roleName + ":::" + tabType + ":::" + tab["tabLabel"].ToString() + ":::" + tab["name"].ToString()] = new JObject
+                {
+                  ["type"] = "string",
+                  ["enum"] = new JArray("true", "false"),
+                  ["x-ms-summary"] = roleName + " - Tab Type: " + tabType + " - Tab label: " + tab["tabLabel"].ToString() + " - Name: " + tab["name"].ToString() +" Selected"
+                };
+              }
+              else
+              {
+                itemProperties[roleName + ":::" + tabType + ":::" + tab["tabLabel"].ToString()] = new JObject
+                {
+                  ["type"] = "string",
+                  ["x-ms-summary"] = roleName + " - Tab Type: " + tabType + " - Tab label: " + tab["tabLabel"].ToString()
+                };
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private void GenerateCustomFields(JObject body, JObject itemProperties)
```

Comment:
There is lot of code repeating in this method. Please optimize.

Reply:
@youngl-docusign You missed this comment.

Reply:
@harshitav-docusign The same reason as the comment above. Link to comment: [comment link](https://github.com/release-docusign/PowerPlatformConnectors/pull/41/files#r1882926598).